configure.ac

```diff
@@ -143,10 +143,19 @@ AC_SEARCH_LIBS([fi_getinfo], [fabric], [], [
   AC_MSG_ERROR([unable to find libfabric])
 ])
 
+AC_ARG_ENABLE([picky-compiler],
+   [AS_HELP_STRING([--disable-picky-compiler], [Disable adding picky compiler flags.])])
+AS_IF([test "${enable_picky_compiler}" != "no"],
+      [picky_compiler_flags="-Wall -Wno-sign-compare"
```

Comment:
If we are using a picky compiler, we should enable sign comparison, right?

Reply:
Yes.  This is what we used to have (removed from the Makefile.am below) and I was trying to minimize changes.  But currently there are no warnings when we do check for signed compare, so I made the change.